ssm_cache/cache.py

```diff
@@ -35,50 +32,89 @@ def _should_refresh(self):
             return True
         # force refresh only if max_age seconds have expired
         return datetime.utcnow() > self._last_refresh_time + self._max_age_delta
-
+    
     def refresh(self):
-        """ Force refresh of the configured param names """
-        response = self.ssm_client.get_parameters(
-            Names=self._names,
-            WithDecryption=self._with_decryption,
-        )
-        # create a dict of name:value for each param
-        self._values = {
-            param['Name']: param['Value']
-            for param in response['Parameters']
-        }
+        self._refresh()
         # keep track of update date for max_age checks
         self._last_refresh_time = datetime.utcnow()
 
-    def value(self, name=None):
+
+class SSMParameterGroup(Refreshable):
+    def __init__(self, max_age=None, with_decryption=True):
+        super(SSMParameterGroup, self).__init__(max_age)
+        
+        self._with_decryption = with_decryption
+        self._parameters = {}
+    
+    def parameter(self, *args, **kwargs):
+        kwargs = kwargs.copy()
+        if 'max_age' in kwargs:
+            raise ValueError("max_age can't be set individually for grouped parameters")
+        if 'with_decryption' not in kwargs:
+            kwargs['with_decryption'] = self._with_decryption
+        parameter = SSMParameter(*args, **kwargs)
+        parameter._group = self
+        self._parameters[parameter._name] = parameter
+        return parameter
+    
+    def _refresh(self):
+        # use batch get
+        with_decryption = any(p._with_decryption for p in six.itervalues(self._parameters))
+        names = [p._name for p in six.itervalues(self._parameters)]
+        invalid_names = []
+        for name_batch in _batch(names, 10): # can only get 10 parameters at a time
+            response = self.ssm_client.get_parameters(
+                Names=name_batch,
+                WithDecryption=with_decryption,
+            )
+            invalid_names.extend(response['InvalidParameters'])
+            for item in response['Parameters']:
+                self._parameters[item['Name']]._value = item['Value']
+        if invalid_names:
+            raise InvalidParam(",".join(invalid_names))
+
+class SSMParameter(Refreshable):
+    """ The class wraps an SSM Parameter and adds optional caching """
+
+    def __init__(self, param_name, max_age=None, with_decryption=True):
+        super(SSMParameter, self).__init__(max_age)
+        if not param_name:
+            raise ValueError("Must specify name")
+        self._name = param_name
+        self._value = None
+        self._with_decryption = with_decryption
+        self._group = None
+
+    def _refresh(self):
+        """ Force refresh of the configured param names """
+        if self._group:
+            return self._group.refresh()
```

Comment:
I would not expect this from the library, as I'm manually forcing the refresh of an individual parameter (not the entire group).

Reply:
Also, it looks like this parameter would be refreshed twice. Or I'm missing something?